.github/workflows/build-test-distribute.yaml

```diff
@@ -71,34 +71,55 @@ jobs:
           make dev/tools
       - run: |
           make test
-  distributions:
-    timeout-minutes: 40
-    needs: ["check", "test", "test_e2e", "test_e2e_env"]
-    if: ${{ always() }}
+  metadata:
+    name: Metadata
+    needs: ["test", "check"]
     runs-on: ubuntu-latest
+    outputs:
+      ENALED_GOARCHES: ${{ steps.set-full-matrix-switches.outputs.ENABLED_GOARCHES }}
+      ENABLED_GOOSES: ${{ steps.set-full-matrix-switches.outputs.ENABLED_GOOSES }}
+      ALLOW_PUSH: ${{ steps.artifacts_metadata.outputs.ALLOW_PUSH }}
     steps:
-      - name: "Halt due to previous failures"
-        if: ${{ contains(needs.*.result, 'failure')|| contains(needs.*.result, 'cancelled') }}
-        run: |
-          exit 1
-          # for some reason, GH Action will always trigger a downstream job even if there are errors in an dependent job
-          # so we manually check it here. An example could be found here: https://github.com/kumahq/kuma/actions/runs/7044980149
       - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
         with:
           fetch-depth: 0
       - name: "Maybe set full matrix"
         if: github.event_name == 'push' || contains(github.event.pull_request.labels.*.name, 'ci/run-full-matrix')
         id: set-full-matrix-switches
         run: |
-          echo 'ENABLED_GOARCHES=arm64 amd64' >> $GITHUB_ENV
-          echo 'ENABLED_GOOSES=linux darwin' >> $GITHUB_ENV
+          echo 'ENABLED_GOARCHES=arm64 amd64' >> $GITHUB_OUTPUT
+          echo 'ENABLED_GOOSES=linux darwin' >> $GITHUB_OUTPUT
       - name: "Add matrix to .run-full-matrix for cache"
         run: |
           echo '${ENABLED_GOARCHES}|${ENABLED_GOOSES}' > .run-full-matrix
       - name: "Maybe set flag to push build artifacts"
+        id: artifacts_metadata
         if: github.event_name == 'push' || contains(github.event.pull_request.labels.*.name, 'ci/force-publish')
         run: |
-          echo 'ALLOW_PUSH=true' >> $GITHUB_ENV
+          echo 'ALLOW_PUSH=true' >> $GITHUB_OUTPUT
+  build:
+    needs: ["metadata", "test", "test_e2e", "test_e2e_env"]
```

Comment:
I'd have this only depend on `check` so it'd make the build generally quicker as build would be in parallel

Reply:
fixed to depend on `["check"]` to make build job in parallel with tests